external_deps/build.sh

```diff
@@ -674,6 +674,8 @@ build_wipe() {
 
 # Common setup code
 common_setup() {
+	HOST="${2}"
+	"common_setup_${1}"
```

Comment:
I added a commit that de-factorize that but I really dislike that as it makes it much verbose and require more boiler plate when adding platforms, making it more error prone.

Reply:
This more-verbose-more-boilerplate-more-copypaste-error-prone-implementation was one of my early factorization iteration I factorized over again, and was only meant to be a temporary step in a step-by-step factorization effort, I would consider merging that extra commit as a bad.

This is a build script, this can suffer an extra indirection if that removes boilerplate copypaste. With indirection, one cannot add an extra platdform and forget to write the platform parent without getting an obvious error yelled by the bash, while with the boilerplate code, one can add a platform parent and forget to call `common_setup` and get unobvious errors instead.

Reply:
I removed the `more-verbose-more-boilerplate-more-copypaste-error-prone-implementation` commit because In find this stupid to add more than 10 lines of copypaste just to avoid an extra level of indirection in a build script that is likely used by two people once in a year…

Reply:
> I removed the `more-verbose-more-boilerplate-more-copypaste-error-prone-implementation` commit

RIP, I never got to see it. But it's not a super important issue so LGTM